Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Document/ShaderManagementConsoleDocument.cpp

```diff
@@ -133,6 +135,68 @@ namespace ShaderManagementConsole
         SetShaderVariantListSourceData(newSourceData);
     }
 
+    void ShaderManagementConsoleDocument::DefragmentVariantList()
+    {
+        struct VariantCompacterKey
+        {
+            AZ::RPI::ShaderVariantListSourceData::VariantInfo* info;
+            size_t hash;
```

Comment:
I'm not too keen on doing this because it adds a default constructor to a POD. With C++ odd rules I'm not sure how far it brings us to need to respect Meyer's "rule of five defaults". This is a local struct with a static "Make" function that can verifiably with a Ctrl+F be found that is the only place where that struct is created.